src/migration/MigrationExecutor.ts

```diff
@@ -424,52 +443,139 @@ export class MigrationExecutor {
+        // so that we have one centralized place that controls this behavior
+
+        // When transaction mode is `each` the default is to run in a transaction
+        // When transaction mode is `none` the default is to not run in a transaction
+        // When transaction mode is `all` the default is to not run in a transaction
+        // since all the migrations are already running in one single transaction
+
+        const txModeDefault = {
+            each: true,
+            none: false,
+            all: false,
+        }[this.transaction]
+
+        for (const migration of migrationsToRevert) {
+            if (migration.instance) {
+                const instanceTx = migration.instance.transaction
+
+                if (instanceTx === undefined) {
+                    migration.transaction = txModeDefault
+                } else {
+                    migration.transaction = instanceTx
+                }
+            }
+        }
 
         // start transaction if its not started yet
         let transactionStartedByUs = false
-        if (this.transaction !== "none" && !queryRunner.isTransactionActive) {
+        if (this.transaction === "all" && !queryRunner.isTransactionActive) {
+            await queryRunner.beforeMigration()
             await queryRunner.startTransaction()
             transactionStartedByUs = true
         }
 
+        // revert the migrations sequentially
         try {
-            if (!this.fake) {
-                await queryRunner.beforeMigration()
-                await migrationToRevert.instance!.down(queryRunner)
-                await queryRunner.afterMigration()
-            }
+            for (const migration of migrationsToRevert) {
+                if (this.fake) {
+                    await this.deleteExecutedMigration(queryRunner, migration)
 
-            await this.deleteExecutedMigration(queryRunner, migrationToRevert)
-            this.connection.logger.logSchemaBuild(
-                `Migration ${migrationToRevert.name} has been ${
-                    this.fake ? "(fake)" : ""
-                } reverted successfully.`,
-            )
+                    continue
+                }
+
+                if (migration.transaction && !queryRunner.isTransactionActive) {
+                    await queryRunner.beforeMigration()
+                    await queryRunner.startTransaction()
+                    transactionStartedByUs = true
+                }
+
+                await migration.instance!.down(queryRunner).catch((error) => {
+                    this.connection.logger.logMigration(
+                        `Reverting of "${migration.name}" failed, error: ${error?.message}`,
+                    )
+                    throw error
+                })
+
+                await this.deleteExecutedMigration(queryRunner, migration)
+
+                // commit transaction if we started it
+                if (migration.transaction && transactionStartedByUs) {
```

Comment:
```suggestion
                if (migration.transaction === 'each' && transactionStartedByUs) {
```

Reply:
`migration.transaction` is a boolean that can be set on the `Migration` class itself:
https://typeorm.io/migrations#transaction-modes

(Not the transaction mode set in `MigrationExecutor`)